test/e2e/operator_condition_e2e_test.go

```diff
@@ -134,34 +117,39 @@ var _ = Describe("Operator Condition", func() {
 
 		// Get the OperatorCondition for csvA and report that it is upgradeable, unblocking csvB
 		upgradeableTrueCondition := metav1.Condition{
-			Type:               operatorsv1.Upgradeable,
+			Type:               operatorsv2.Upgradeable,
 			Status:             metav1.ConditionTrue,
 			Reason:             "test",
 			Message:            "test",
 			LastTransitionTime: metav1.Now(),
 		}
 		Eventually(func() error {
-			cond, err = crc.OperatorsV1().OperatorConditions(testNamespace).Get(context.TODO(), csvA.GetName(), metav1.GetOptions{})
-			if err != nil {
+			cond, err = crc.OperatorsV2().OperatorConditions(testNamespace).Get(context.TODO(), csvA.GetName(), metav1.GetOptions{})
+			if err != nil || currentGen == cond.ObjectMeta.GetGeneration() {
 				return err
 			}
-			meta.SetStatusCondition(&cond.Status.Conditions, upgradeableTrueCondition)
-			_, err = crc.OperatorsV1().OperatorConditions(testNamespace).UpdateStatus(context.TODO(), cond, metav1.UpdateOptions{})
+			currentGen = cond.ObjectMeta.GetGeneration()
+			upgradeableTrueCondition.ObservedGeneration = cond.ObjectMeta.GetGeneration()
+			meta.SetStatusCondition(&cond.Spec.Conditions, upgradeableTrueCondition)
+			_, err = crc.OperatorsV2().OperatorConditions(testNamespace).Update(context.TODO(), cond, metav1.UpdateOptions{})
 			return err
 		}, pollInterval, pollDuration).Should(Succeed())
 
 		// Await csvB's success
 		_, err = awaitCSV(crc, testNamespace, csvB.GetName(), csvSucceededChecker)
 		require.NoError(GinkgoT(), err)
 
-		// Get the OperatorCondition for csvB and report that it is not upgradeable
+		// Get the OperatorCondition for csvB and purposedly change ObservedGeneration
+		// to cause mismatch generation situation
 		Eventually(func() error {
-			cond, err = crc.OperatorsV1().OperatorConditions(testNamespace).Get(context.TODO(), csvB.GetName(), metav1.GetOptions{})
-			if err != nil {
+			cond, err = crc.OperatorsV2().OperatorConditions(testNamespace).Get(context.TODO(), csvB.GetName(), metav1.GetOptions{})
+			if err != nil || currentGen == cond.ObjectMeta.GetGeneration() {
 				return err
 			}
-			meta.SetStatusCondition(&cond.Status.Conditions, upgradeableFalseCondition)
-			_, err = crc.OperatorsV1().OperatorConditions(testNamespace).UpdateStatus(context.TODO(), cond, metav1.UpdateOptions{})
+			currentGen = cond.ObjectMeta.GetGeneration()
+			upgradeableTrueCondition.ObservedGeneration = currentGen + 1
+			meta.SetStatusCondition(&cond.Status.Conditions, upgradeableTrueCondition)
```

Comment:
Just to confirm: you chose to change the `cond.Status.Condition` entry because doing so in the Spec would just cause OLM to observe a new generation eh?

Reply:
Correct. Updating Status won't increment generation.